rawx/notifier.go

```diff
@@ -81,38 +75,61 @@ func makeBeanstalkNotifier(endpoint string,
 	notifier.endpoint = endpoint
 	notifier.tube = beanstalkNotifierDefaultTube
 	// TODO(adu) Check endpoint
+	notifier.beanstalkd = nil
 	return notifier, nil
 }
 
-func (notifier *beanstalkNotifier) start() {
+func (notifier *beanstalkNotifier) Start() {
 	// A single notifier by goroutine.
 	notifier.wg.Wait()
 
 	notifier.wg.Add(1)
 	go func() {
 		defer notifier.wg.Done()
 		for eventJSON := range notifier.queue {
-			if err := notifier.syncNotify(eventJSON); err != nil {
-				LogWarning("Notify error: %s", err)
-			}
+			notifier.syncNotify(eventJSON)
 		}
 	}()
 }
 
-func (notifier *beanstalkNotifier) stop() {
+func (notifier *beanstalkNotifier) Stop() {
 	close(notifier.queue)
```

Comment:
Careful here no guarantees when you close the channel that an `asyncNotify` is not trying to write in the channel, which will crash the goroutine.